app/pages/app/app.js

```diff
@@ -343,6 +344,34 @@ export class AppComponent extends React.Component {
     return null;
   }
 
+  renderNavPatientHeader() {
+    const { 
+      patient, 
+      user,
+      permsOfLoggedInUser,
+      context: { trackMetric },
+      clinicFlowActive,
+      selectedClinicId,
+      location: currentPage,
+      query,
+    } = this.props;
+
+    if (!this.isPatientVisibleInNavbar()) return null; // only show on pages with a patient of focus
```

Comment:
This _seems_ to be the pertinent condition, lifted from `app/pages/app/app.js:317`, but I'm not 100% certain. Your eyes would be appreciated